sharc/propagation/propagation_input_file.py

```diff
@@ -0,0 +1,202 @@
+                upper_right = [float(x) for x in head_list[idx_list[6] + 1:
+                                                           idx_list[6] + 3]]
+                height = float(head_list[idx_list[7] + 1])
+                resolution = float(head_list[idx_list[8] + 1])
+                receive_gain = float(head_list[idx_list[9] + 1])
+
+                # Create header struct
+                head = PathLossHeader(antenna,
+                                      location,
+                                      frequency,
+                                      power,
+                                      antenna_type,
+                                      lower_left,
+                                      upper_right,
+                                      height,
+                                      resolution,
+                                      receive_gain)
+
+                # Initialize path loss array
+                n_lin = int((head.upper_right[1] -
+                         head.lower_left[1])/head.resolution)
+                n_col = int((head.upper_right[0] -
+                         head.lower_left[0])/head.resolution)
+                loss = -np.inf*np.ones((n_lin,n_col))
+
+                # Loop through all the remaining lines
+                line = next(f)
+                while "END_DATA" not in line:
+                    data = [float(x) for x in line.split()]
+
+                    # Define line and column of array
+                    lin = int((data[1] - (head.lower_left[1] +
+                           head.resolution/2))/head.resolution)
+                    col = int((data[0] - (head.lower_left[0] +
+                           head.resolution/2))/head.resolution)
+                    # Invert signal to match simulator convention
+                    loss[lin, col] = (-1)*(data[2] + head.receiver_gain)
+
+                    line = next(f)
+
+                # Add values to dict
+                self.path_loss[head.antenna] = (head, loss)
+
+    def get_loss(self, *args, **kwargs) -> np.array:
+        """
+        Returns from given BSs in the specified UE positions
+
+        Keyword Arguments:
+            bs_id (np.array): array of strings with IDs of BSs to which
+                calculate the path loss
+            ue_position_x (np.array): array of x coordinate of UEs to
+                which calculate path loss
+            ue_position_y (np.array):array of y coordinate of UEs to
+                which calculate path loss
+
+        Returns:
+            loss (np.array): path loss matrix. Lines correspond to BSs while
+                columns correspond to UEs
+        """
+        # Get keywork arguments
+        bs_id = kwargs["bs_id"]
+        ue_position_x = kwargs["ue_position_x"]
+        ue_position_y = kwargs["ue_position_y"]
+
+        # Initialize array
+        loss = np.zeros((len(bs_id),len(ue_position_x)))
+
+        # Loop through all the cells
+        for k in range(len(bs_id)):
```

Comment:
Is `bs_id` a list of Base stations ids? If so, use `enumerate(bs_id)` to iterate over bs_id elements and indices.